client/tx/factory.go

```diff
@@ -57,15 +57,35 @@ func NewFactoryCLI(clientCtx client.Context, flagSet *pflag.FlagSet) Factory {
 		signMode = signing.SignMode_SIGN_MODE_EIP_191
 	}
 
+	var gasSetting client.GasSetting
+	gasStr, err := flagSet.GetString(flags.FlagGas)
+	if err != nil {
```

Comment:
just to refresh my memory here, `err` is non-nil when the value is NOT provided, right?

Reply:
right, in this case we use the value in `client.toml`

Reply:
> just to refresh my memory here, `err` is non-nil when the value is NOT provided, right?

actually no, err is nil even if the flag is not provided. this breaks the logic, so i'm converting this PR back to draft again before this is fixed

Reply:
`err` is only non-nil under 3 situations:
* the flag is not defined
* the flag is defined as a different data type
* parsing the input fails

https://github.com/spf13/pflag/blob/v1.0.5/flag.go#L375-L384